docs/developer/developer-handbook.md

```diff
@@ -0,0 +1,93 @@
+# EDC Developer Handbook
+
+<!-- TOC -->
+* [EDC Developer Handbook](#edc-developer-handbook)
+  * [Introduction](#introduction)
+  * [Building a distribution](#building-a-distribution)
+    * [Seed some data](#seed-some-data)
+    * [Perform a data transfer](#perform-a-data-transfer)
+  * [The control plane - core concepts](#the-control-plane---core-concepts)
+    * [API objects in detail](#api-objects-in-detail)
+    * [EDC state machines](#edc-state-machines)
+    * [The extension model](#the-extension-model)
+    * [EDC dependency injection](#edc-dependency-injection)
+    * [Policy scopes and evaluation](#policy-scopes-and-evaluation)
+  * [The data plane](#the-data-plane)
+    * [Data plane selectors](#data-plane-selectors)
+    * [Writing a DataSink and DataSource extension](#writing-a-datasink-and-datasource-extension)
+    * [The Control API](#the-control-api)
+  * [Advanced concepts](#advanced-concepts)
+    * [Events and callbacks](#events-and-callbacks)
+    * [The EDC JUnit framework](#the-edc-junit-framework)
+    * [Automatic documentation](#automatic-documentation)
+    * [Customize the build](#customize-the-build)
+<!-- TOC -->
+
+## Introduction
+
+--> states the intended audience, required pre-existing knowledge, required tools, etc.
+
+## Building a distribution
+--> link to one of the samples
+
+### Seed some data
+
+--> use postman, or even just curl
+
+### Perform a data transfer
+--> actually launches another connector, shows a data exchange -> link to samples
+
+## The control plane - core concepts
+
+### API objects in detail
+
+--> explains Assets, Policies, Contract Definitions, etc. from an external API perspective. mentions JSON-LD and related
+specs (ODRL, DCAT)
+
+### EDC state machines
+
+--> gives an overview of the state machines to understand asynchronicity and why some APIs only return IDs
+
+### The extension model
```

Comment:
Would be great to have an overview of all extension points (at least groups)

Reply:
there is a lot of them, and we can already convert the autodoc manifest to markdown, so I would link that MD file

Reply:
There are many indeed, and so an overview might be useful for developers. The suggested link to the manifest should work well. Maybe accompanied by mentioning the basic categories like Authentication & Security, Asset. Negotiation, Transfer, Policy Engine, Protocol?

Reply:
How about this: we outline what "plugpoints" we have, i.e. where devs can hook in and extend/customize their runtimes? Those would include:
- Authentication/Authorization of the Management API
- policies and evaluation functions
- persistence implementations
- provisioners
- transaction management (?)

Replacing the Protocol (DSP) is a _really_ advanced topic, which I would not put in the developer handbook.